src/App.jsx

```diff
@@ -14,33 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const FILTER_TYPE_ALPHABET = 'alphabetically';
+const FILTER_TYPE_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getPreparedGoods = (goods, sortOptions) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortOptions.filterType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortOptions.filterType) {
+        case FILTER_TYPE_ALPHABET:
+          return good1.localeCompare(good2);
+        case FILTER_TYPE_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortOptions.isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [filterInstruction, setFilterInstruction] = useState({
+    filterType: '',
+    isReversed: false,
+  });
+
+  const isAnyFilterApplied =
+    filterInstruction.filterType || filterInstruction.isReversed;
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, filterInstruction);
+
+  const handleSetFilterType = filterTypeName => {
+    setFilterInstruction(prevState => {
+      return {
+        ...prevState,
+        filterType: filterTypeName,
+      };
+    });
+  };
+
+  const handleReverse = () => {
+    setFilterInstruction(prevState => {
+      return {
+        ...prevState,
+        isReversed: !prevState.isReversed,
+      };
+    });
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleClearFilter = () => {
+    setFilterInstruction(prevState => {
+      return {
+        ...prevState,
+        filterType: '',
+        isReversed: false,
+      };
+    });
+  };
+
+  const isLightOn = filterType => {
+    return filterInstruction.filterType !== filterType;
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => handleSetFilterType(FILTER_TYPE_ALPHABET)}
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': isLightOn(FILTER_TYPE_ALPHABET),
+          })}
```

Comment:
Thats so strange why I have errors in tests trying to push my code with 'cn' library...